c7n/filters/offhours.py

```diff
@@ -361,8 +360,8 @@ def get_default_schedule(self):
 class OnHour(Time):
 
     schema = type_schema(
-        'onhour', rinherit=Time.schema, required=['onhour', 'default_tz'],
-        onhour={'type': 'integer', 'minimum': 0, 'maximum': 24})
+        'onhour', rinherit=Time.schema, required=['default_tz'],
```

Comment:
remove the requirement for setting the default onhour time. You can still set it an it works, but you can not set it. This gives folks to option to use the policy so each user has to explicitly set the offhours/onhours, and if they set an on or offhours tag with no value, there will no be default action that is taken. I personally like having the user be more explicit about opting into this policy, this way the user isn't surprised when actions happen and they're not sure why.